include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy_impl.h

```diff
@@ -77,7 +77,22 @@ namespace experimental{
 
     template<typename ...Args>
     selection_handle_t select(Args&&...) {
-      auto i = unit_->next_context_++ % unit_->num_contexts_;
+      size_t i=0;
+      while(true){
+          universe_container_size_t current_context_ = unit_->next_context_.load();
```

Comment:
Looks fine now. Not a huge fan of unit_ as the name but its an implementation detail, so it's ok. Eventually, unit_ should be private though. But that might wait to refactoring with new names, etc.